pandas/io/json.py

```diff
@@ -883,3 +909,156 @@ def _recursive_extract(data, path, seen_meta, level=0):
         result[k] = np.array(v).repeat(lengths)
 
     return result
+
+
+# TODO: Make method on Timedelta?
+def isoformat(x):
+    """
+    Format Timedelta as ISO 8601 Duration
+
+    Examples
+    --------
+    >>> td = pd.Timedelta(days=6, minutes=50, seconds=3,
+    ...                   milliseconds=10, microseconds=10, nanoseconds=12)
+    >>> isoformat(td)
+    'Pn6Tn0n50n3.010010012'
+    """
+    components = x.components
+    seconds = '{}.{:0>3}{:0>3}{:0>3}'.format(components.seconds,
+                                             components.milliseconds,
```

Comment:
yeah let's do this in a separate PR 
ideally also be able to parse this as well (open an issue of u don't do it in same PR)